services/web3id-issuer/src/main.rs

```diff
@@ -0,0 +1,376 @@
+                (
+                    StatusCode::BAD_REQUEST,
+                    axum::Json(format!("Credential expiry is too early.")),
+                )
+            }
+        };
+        r.into_response()
+    }
+}
+
+#[derive(Clone, Debug)]
+struct State {
+    client:             Cis4Contract,
+    issuer:             Arc<WalletAccount>,
+    nonce_counter:      Arc<tokio::sync::Mutex<Nonce>>,
+    // In milliseconds
+    min_allowed_expiry: u64,
+}
+
+#[derive(Debug, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct IssueRequest {
+    credential: CredentialInfo,
+    #[serde(deserialize_with = "concordium_base::common::base16_decode")]
+    signature:  [u8; 64],
+    data:       DataToSign,
+}
+
+#[derive(Debug, serde::Serialize)]
+#[serde(tag = "status")]
+enum StatusResponse {
+    #[serde(rename = "finalized")]
+    Finalized { block: BlockHash, success: bool },
+    #[serde(rename = "notFinalized")]
+    NotFinalized,
+}
+
+#[tracing::instrument(level = "info", skip(state, tx))]
+async fn status(
+    axum::extract::State(mut state): axum::extract::State<State>,
+    tx: Result<axum::extract::Path<TransactionHash>, PathRejection>,
+) -> Result<axum::Json<StatusResponse>, Error> {
+    let tx = tx?;
+    let status = state.client.client.get_block_item_status(&tx).await?;
+    if let Some((bh, summary)) = status.is_finalized() {
+        Ok(axum::Json(StatusResponse::Finalized {
+            block:   *bh,
+            success: summary.is_success(),
+        }))
+    } else {
+        Ok(axum::Json(StatusResponse::NotFinalized))
+    }
+}
+
+#[tracing::instrument(level = "info", skip(state, request))]
+async fn issue_credential(
+    axum::extract::State(mut state): axum::extract::State<State>,
+    request: Result<axum::Json<IssueRequest>, JsonRejection>,
+) -> Result<axum::Json<TransactionHash>, Error> {
+    tracing::info!("Request to issue a credential.");
+    let axum::Json(request) = request?;
+    if request.data.timestamp.timestamp_millis()
+        < (chrono::Utc::now().timestamp_millis() as u64).saturating_add(state.min_allowed_expiry)
+    {
+        return Err(Error::ExpiryTooEarly);
+    }
+
+    let expiry = TransactionTime::from_seconds(request.data.timestamp.timestamp_millis() / 1000);
+
+    let storage_data = concordium_std::to_bytes(&StoreParam {
+        public_key: request.credential.holder_id,
+        signature:  request.signature,
+        data:       request.data,
+    });
+
+    let mut nonce_guard = state.nonce_counter.lock().await;
+    tracing::info!("Using nonce {} to send the transaction.", *nonce_guard);
+    let metadata = Cis4TransactionMetadata {
+        sender_address: state.issuer.address,
+        nonce: *nonce_guard,
+        expiry,
+        energy: GivenEnergy::Add(10_000.into()),
+        amount: Amount::zero(),
+    };
+
+    let tx = state
+        .client
+        .register_credential(
+            &*state.issuer,
```

Comment:
Because state.issuer has type `Arc<WalletAccount>` but what we need is `&WalletAccount`.

This is making use of `Deref` (https://doc.rust-lang.org/std/ops/trait.Deref.html) to get a reference to `&WalletAccount` out of the Arc.

So there is a bit of implicit compiler builtin stuff going on.